eng/common/pipelines/templates/jobs/spec-gen-sdk.yml

```diff
@@ -0,0 +1,210 @@
+parameters:
+  - name: SpecRepoCommit
+    type: string
+    default: 'HEAD'
+  - name: SdkRepoCommit
+    type: string
+    default: 'HEAD'
+  - name: ConfigType
+    type: string
+    values:
+      - 'TypeSpec'
+      - 'OpenAPI'
+    default: 'TypeSpec'
+    displayName: 'Configuration Type'
+  - name: ConfigPath
+    type: string
+    default: 'specification/contosowidgetmanager/Contoso.Management/tspconfig.yaml'
+    displayName: 'Path to TypeSpec or OpenAPI Configuration file'
+  - name: ApiVersion
+    type: string
+    default: 'not-specified'
+    displayName: 'API Version of Spec to be used for SDK Generation'
+  - name: SkipPullRequestCreation
+    type: boolean
+    default: false
+
+variables:
+  SdkRepoUrl: $(SdkRepoUrlForLanguage) # This variable is set in the pipeline UI for each language e.g. https://github.com/Azure/azure-sdk-for-net for dotnet
+
+jobs:
+- job:
+  displayName: 'SDK Generation'
+
+  variables:
+    - template: /eng/pipelines/templates/variables/image.yml
+    - name: SpecRepoUrl
+      value: 'https://github.com/$(Build.Repository.Name)'
+    - name: NodeVersion
+      value: '22.13.x'
+    - name: PythonVersion
+      value: '3.13'
+    - name: SpecGenSdkVersion
+      value: 'latest'
+    - name: SdkArtifactName
+      value: SDK_Artifact
+
+  pool:
+    name: $(LINUXPOOL)
+    vmImage: $(LINUXVMIMAGE)
+
+  steps:
+    - checkout: none
+
+    - pwsh: |
+        $urlPattern = '^https://github\.com/(?<organization>[^/]+)/(?<repository>[^/]+)'
+        if ('$(SpecRepoUrl)' -match $urlPattern) {
+          $specRepoOwner = $Matches['organization']
+          Write-Host "##vso[task.setvariable variable=SpecRepoOwner]$specRepoOwner"
+          Write-Host "SpecRepoOwner variable set to: $specRepoOwner"
+
+          $specRepoName = $Matches['repository']
+          Write-Host "##vso[task.setvariable variable=SpecRepoName]$specRepoName"
+          Write-Host "SpecRepoName variable set to: $specRepoName"
+
+          $specRepoDirectory = "$(System.DefaultWorkingDirectory)/$specRepoName"
+          Write-Host "##vso[task.setvariable variable=SpecRepoDirectory]$specRepoDirectory"
+          Write-Host "SpecRepoDirectory variable set to: $specRepoDirectory"
+        }
+
+        if ('$(SdkRepoUrl)' -match $urlPattern) {
```

Comment:
Could you pass in $(SdkRepoUrl) by parameter to have better readability?

Reply:
I agree this should be a template parameter but it is more about flexibility for me. It will enable us to reuse this template in other contexts that I believe will come up.

Reply:
The reason for using a variable is to make it so user cannot change this at queue time. We only wanted them to be able to override the org so they can also run the pipeline from their fork. So that why this is a variable, at least that my understanding from my last discussion with @raych1 .

Reply:
As discussed offline, we will use this yml file as a base template then create a single yml file in spec repo to reference it. With this way, $SdkRepoUrl is not exposed to user when queuing the run, but we keep the flexibility to override it when this yml file is referenced by other pipeline yml files.